### PR TITLE
Wip/mehrad/kubeflow

### PR DESCRIPTION
**Pull request by Mehrad0711**
Most changes in this PR are for SPL pipelines: fixing things and adding new features. 

Some of the images for thingtalk2 come with preinstalled genie-toolkit and thingtalk, and running rebuild fails. I changed Dockerfile to account for that but not sure if there is a better way instead of `if else`.

### CODE REVIEW

**Comment by gcampax** [comment]
I'm not sure I understand what the problem is with the existing docker file. Are you trying to use an existing image as base image for a rebuild? That doesn't make sense.

File: pipelines/common.py [hunk 1]
```diff
@@ -35,7 +35,7 @@
 GENIENLP_VERSION = 'd04ed4a2c38788eab9a9f4694a20fddeba62ea7d'
 GENIE_VERSION = '862f3444aaee0522c84aa7b24ad0a3f7203b9f48'
 THINGTALK_VERSION = 'a3eb276cab0f554646ee6ef5620be12179f55ba7'
-BOOTLEG_VERSION = 'f53e67397ddcd099f3a18a014c9ce82b02d2223c'
+BOOTLEG_VERSION = 'master'
```
Comment (gcampax) [inline pipelines/common.py]: You need to use hashes in pipeline parameters, we already talked about this.

File: sync-repos.sh [hunk 2]
```diff
@@ -44,7 +46,7 @@ if [ -n "${GENIE_VERSION}" ] && [ "${GENIE_VERSION}" != "${GENIE_HEAD}" ]; then
   git fetch
   git checkout -f ${GENIE_VERSION}
   if test -f yarn.lock ; then
-    yarn link thingtalk
+#    yarn link thingtalk
```
Comment (gcampax) [inline sync-repos.sh]: This line is wrong, and also should have no effect on recent versions of genie that have switched to npm. On versions of genie that use yarn, it's necessary or nothing works.

**Comment by Mehrad0711** [comment]
Yes. If we are always supposed to start from the base image then we won't need to change. But if you just wanna bump the version for one repo it's faster to rebuild on top of an existing image. It's just for speed but it's not needed so I can drop it.

**Comment by gcampax** [comment]
Yeah it's better not confuse things by doing rebuilds on top of rebuilds, please drop.

**Pull request merged**